src/Mod/Start/App/DisplayedFilesModel.cpp

```diff
@@ -38,30 +38,24 @@ using namespace Start;
 namespace
 {
 
-std::string humanReadableSize(unsigned int bytes)
+static std::string humanReadableSize(const size_t size)
 {
-    static const std::vector<std::string> siPrefix {
-        "b",
-        "kb",
-        "Mb",
-        "Gb",
-        "Tb",
-        "Pb",
-        "Eb"  // I think it's safe to stop here (for the time being)...
-    };
-    size_t base = 0;
-    double inUnits = bytes;
-    constexpr double siFactor {1000.0};
-    while (inUnits > siFactor && base < siPrefix.size() - 1) {
-        ++base;
-        inUnits /= siFactor;
-    }
-    if (base == 0) {
-        // Don't include a decimal point for bytes
-        return fmt::format("{:.0f} {}", inUnits, siPrefix[base]);
-    }
-    // For all others, include one digit after the decimal place
-    return fmt::format("{:.1f} {}", inUnits, siPrefix[base]);
+    static constexpr std::array<std::pair<char, unsigned long long>, 7> prefixes {
+        {{0, 0},
+         {'k', 1ULL << 10},
+         {'M', 1ULL << 20},
+         {'G', 1ULL << 30},
+         {'T', 1ULL << 40},
+         {'P', 1ULL << 50},
+         {'E', 1ULL << 60}}};
+
+    const auto spec = std::find_if(prefixes.rbegin(), prefixes.rend(), [&](const auto& pair) {
+        return pair.second <= size;
+    });
+
+    return spec->first == 0

```

Comment:
`spec->first == 0` is not helpful, what does this condition mean? That should be obvious just by looking at the condition. If you use struct instead of std::pair you can express it like `!spec->prefix` which would be meaningful.

Reply:
This is where the comment @chennes was talking about enters the scene. As noted above, for a nice descriptive names, etc. someone else must step in and modify code.